server/routes.go

```diff
@@ -189,7 +191,24 @@ func (s *Server) GenerateHandler(c *gin.Context) {
 
 	images := make([]llm.ImageData, len(req.Images))
 	for i := range req.Images {
-		images[i] = llm.ImageData{ID: i, Data: req.Images[i]}
+		if isMllama {
+			data, aspectRatioID, err := imageproc.Preprocess(req.Images[i])
+			if err != nil {
+				c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{"error": "error processing image"})
+				return
+			}
+
+			buf := new(bytes.Buffer)
+			err = binary.Write(buf, binary.LittleEndian, data)
```

Comment:
is `binary` and `binary.LittleEndian` required here (vs just `buf := bytes.NewBuffer(data)`)?

Reply:
`data` is `[]float32`, not `[]byte`, so they're not quite the same. Unfortunately you can't pass a slice of `float32` to `bytes.NewBuffer()`.